### PR TITLE
[SRVCOM-2200] Serverless Operator Overrides all Labels and Annotations on Eventing Namespace

### PR DESCRIPTION
**Pull request by pierDipi**
Fixes JIRA #SRVCOM-2200

## Proposed Changes

- Fix filter namespace stage order
- Add E2E test
- Document the procedure to reproduce the issue and use GitOps

### CODE REVIEW

**Comment by pierDipi** [comment]
I can't do much on the reviewdog / lint error since that's the terminology Argo CD uses for their CRD

File: docs/gitops/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+### Reproduce SRVCOM-2200
+
+- Delete Serverless operator pods
+  ```shell
+  oc delete pods -n openshift-serverless --all
+  ```
+- Verify that:
+  - Argo CD UI still reports `Sync OK`
+  - Argo CD application still reports `Synced` with `oc get application -n openshift-gitops openshift-serverless`
```
Comment (pierDipi) [inline docs/gitops/README.md:48]: This will fail you try to do it before my PR is merged

File: docs/gitops/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+## OpenShift Serverless with GitOps
```
Comment (matzew) [inline docs/gitops/README.md:1]: thanks for the instructions, really good

**Comment by matzew** [comment]
I will test-drive this PR with the instructions, thanks!

**Pull request merged**